### PR TITLE
iam export:designate zip file name

### PR DESCRIPTION
**Pull request by dormanze**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## How to test this PR?

image

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by klauspost** [comment]
It seems a bit contrary to the usual design that the parameter is always assumed to be local path and not an alias.

Adding a `-o=/tmp/myminio-iam.zip` that allows to override the output path seems more appropriate to me.

File: cmd/admin-cluster-iam-export.go [hunk 1]
```diff
@@ -32,13 +32,23 @@ import (
 	"github.com/minio/pkg/v3/console"
 )
 
+// iam export specific flags.
+var (
+	iamExportFlags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "overwrite,o",
```
Comment (harshavardhana) [inline cmd/admin-cluster-iam-export.go:39]:
```suggestion
			Name:  "output,o",
```

File: cmd/admin-cluster-iam-export.go [hunk 2]
```diff
@@ -32,13 +32,23 @@ import (
...
+			Usage: "overwrite iam export file path",
```
Comment (harshavardhana) [inline cmd/admin-cluster-iam-export.go:40]:
```suggestion
			Usage: "output iam export to a custom file path",
```

File: cmd/admin-cluster-iam-export.go [hunk 3]
```diff
@@ -52,11 +62,13 @@ FLAGS:
 EXAMPLES:
   1. Download all IAM metadata for cluster into zip file.
      {{.Prompt}} {{.HelpName}} myminio
+  2. Download all IAM metadata for cluster into designated zip file.
```
Comment (harshavardhana) [inline cmd/admin-cluster-iam-export.go:65]:
```suggestion
  2. Download all IAM metadata to a custom file.
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/admin-cluster-iam-export.go [hunk 4]
```diff
@@ -52,11 +62,13 @@ FLAGS:
...
   1. Download all IAM metadata for cluster into zip file.
      {{.Prompt}} {{.HelpName}} myminio
+  2. Download all IAM metadata to a custom file.
+     {{.Prompt}} {{.HelpName}} myminio --output /tmp/myminio-iam.zip
 `,
 }
 
 func checkIAMExportSyntax(ctx *cli.Context) {
-	if len(ctx.Args()) != 1 {
+	if len(ctx.Args()) != 1 && len(ctx.Args()) != 2 {
```
Comment (klauspost) [inline cmd/admin-cluster-iam-export.go:71]: Leftover?

**Review by klauspost — APPROVED** [review]

File: cmd/admin-cluster-iam-export.go [hunk 5]
```diff
@@ -52,6 +62,8 @@ FLAGS:
 EXAMPLES:
   1. Download all IAM metadata for cluster into zip file.
      {{.Prompt}} {{.HelpName}} myminio
+  2. Download all IAM metadata to a custom file.
```
Comment (zveinn) [inline cmd/admin-cluster-iam-export.go:65]: Please add some spacing in the help menu .. a single space should do

**Review by zveinn — APPROVED** [review]

**Review by zveinn — APPROVED** [review]

**Comment by klauspost** [comment]
@dormanze I presume it is a script, but could you please turn off the automatic merge with master? It is very noisy and we need to manually re-run tests every time.

**Review by donatello — APPROVED** [review]

**Pull request merged**